### PR TITLE
[DOR-84][AO] add import mandatory vessel details page

### PR DESCRIPTION
**Pull request by arturopala**

### CODE REVIEW

**Comment by arturopala** [comment]
This PR covers adding mandatory vessel details page for import request type of Hold, and requiring mandatory vessel details for export request type of Hold, C1601, C1602

File: test/uk/gov/hmrc/traderservices/journey/TraderServicesFrontendFormatSpec.scala [hunk 1]
```diff
@@ -201,6 +201,17 @@ class TraderServicesFrontendFormatSpec extends UnitSpec {
           )
         )
       )
+      validateJsonFormat(
+        """{"state":"AnswerImportQuestionsMandatoryVesselInfo","properties":{"declarationDetails":{"epu":"123","entryNumber":"100000Z","entryDate":"2020-10-05"},"importQuestionsAnswers":{"requestType":"New","priorityGoods":"LiveAnimals","freightType":"RORO"}}}""",
+        State.AnswerImportQuestionsMandatoryVesselInfo(
+          DeclarationDetails(EPU(123), EntryNumber("100000Z"), LocalDate.parse("2020-10-05")),
+          ImportQuestions(
+            requestType = Some(ImportRequestType.New),
```
Comment (Tasmine10ten) [inline test/uk/gov/hmrc/traderservices/journey/TraderServicesFrontendFormatSpec.scala:209]: Will need to be requestType.hold (the test underneath will also need to be updated from anything other than hold)

Comment (arturopala) [reply test/uk/gov/hmrc/traderservices/journey/TraderServicesFrontendFormatSpec.scala]: fixed

File: conf/messages.en [hunk 2]
```diff
@@ -350,6 +349,10 @@ form.import-questions.hasALVS.no=No
 form.import-questions.freightType.Maritime=Maritime
 form.import-questions.freightType.Air=Air
 form.import-questions.freightType.RORO=Rail, road, or roll-on, roll-off (RORO)
+form.import-questions.vessel-details.vesselName=Vessel name
```
Comment (Tasmine10ten) [inline conf/messages.en:352]: These are largely fine but could you amend the error messages (at present lines 322 & 323)

error.timeOfArrival.hour.invalid-digits=Time of arrival must only contain numbers
error.timeOfArrival.minutes.invalid-digits=Time of arrival must only contain numbers

image

File: conf/messages.en [hunk 3]
```diff
@@ -350,6 +349,10 @@ form.import-questions.hasALVS.no=No
...
+form.import-questions.vessel-details.dateOfArrival=Date of arrival
+form.import-questions.vessel-details.timeOfArrival=Time of arrival
+form.import-questions.vessel-details.timeOfArrival.hint=For example 14:30
```
Comment (Tasmine10ten) [inline conf/messages.en:355]: Needs a comma: 
image

File: app/uk/gov/hmrc/traderservices/controllers/FormFieldMappings.scala [hunk 4]
```diff
@@ -149,7 +149,7 @@ object FormFieldMappings {
 
   val importHasALVSMapping: Mapping[Boolean] = booleanMapping("importHasALVS", "yes", "no")
 
-  val allowedSpecialNameCharacterSet = Set(' ', '/', '\\', '_', '-', '&', '+', '\'', '.')
+  val allowedSpecialNameCharacterSet = Set(' ', '-', '&', '\'', '.')
```
Comment (Tasmine10ten) [inline app/uk/gov/hmrc/traderservices/controllers/FormFieldMappings.scala:152]: Even though the content for this message doesn't include some of the symbols we allowed I think we're alright to leave them in. This is Irene's comment on the spreadsheet 
image

**Review by Tasmine10ten — APPROVED** [review]

**Pull request merged**